### ISSUE TITLE
aws.vpc.DataAwsVpc cannot be used with VPC id without filtering

### ISSUE THREAD

**Issue by matt-usurp** [hashicorp/terraform-cdk#1602]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

```
node: 16.13.0
npm: 8.1.0

cdk: 0.9.4
cdk-tf: 0.9.4
@cdktf/provider-aws: 5.0.33
```

### Affected Resource(s)

* [aws.vpc.DataAwsVpc](https://github.com/hashicorp/cdktf-provider-aws/blob/main/src/vpc/data-aws-vpc.ts)

### Expected Behavior

To be able to use the examples provided on the [`aws_vpc` data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/vpc) documentation:

```hcl
data "aws_vpc" "selected" {
  id = "vpc-1234567"
}
```
And be able to use that with `@cdktf/provider-aws` as follows:

```ts
import * as aws from '@cdktf/provider-aws';

new aws.vpc.DataAwsVpc(this, 'vpc', {
  id: 'vpc-1234567', 
});
```

### Actual Behavior

The property `id` does not exist on the [interface `DataAwsVpcConfig`](https://github.com/hashicorp/cdktf-provider-aws/blob/main/src/vpc/data-aws-vpc.ts#L9-L36).

### Steps to Reproduce

```ts
import * as aws from '@cdktf/provider-aws';

new aws.vpc.DataAwsVpc(this, 'vpc', {
  id: 'vpc-1234567', 
});
```

### Note

This looks like it can be done by using [the AWS API filters](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcs.html) instead, however documentation for `HCL` syntax clearly shows `id` being used which would be "more ideal".

```ts
new aws.vpc.DataAwsVpc(this, 'vpc', {
  filter: [
    {
      name: 'vpc-id',
      values: [
        'vpc-1234567', 
      ],
    }
  ],
});
```

**Comment by jsteinich** [comment]
I took an initial look at fixing this (I've looked into something similar in the past) and it's potentially a bit tricky.
It seems that provider schemas tend to mark `id` as `optional` for most (maybe all) resources when really it doesn't make sense to set it (not sure if Terraform even allows it). We had put some rules in place to try and work around this but they really aren't working at this time (maybe never did fully). If Terraform is ok with setting `id` perhaps the best action is to just listen to the schema and remove custom rules, but if not, we'll need to some research / experimenting to come up with better rules.

**Comment by matt-usurp** [comment]
I assumed that this might be a caveat of code generation or there were some technical issues. I thought I would raise the bug as there wasn't one similar and it can either become something to implement or be resolved as "won't do". 

As mentioned, it's not a blocker as there is a way around it, it's just a slight difference from the documentation.

**Comment by skorfmann** [comment]
@rirze had similar issues described in https://github.com/hashicorp/terraform-cdk/issues/1669

> Data Source `AwsDataLb` / `AwsDataLbListener`
> Function signature in Python does not show `arn` input parameter. It should according to the Terraform AWS spec.

```python
|  __init__(self, scope: constructs.Construct, id: str, *, name: Optional[str] = None, tags: Union[cdktf.IResolvable, Mapping[str, str], NoneType] = None, count: Union[int, float, cdktf.IResolvable, NoneType] = None, depends_on: Optional[Sequence[cdktf.ITerraformDependable]] = None, lifecycle: Optional[cdktf.TerraformResourceLifecycle] = None, provider: Optional[cdktf.TerraformProvider] = None) -> None
 |      Create a new {@link https://www.terraform.io/docs/providers/aws/d/lb.html aws_lb} Data Source.
 |      
 |      :param scope: The scope in which to define this construct.
 |      :param id: The scoped construct ID. Must be unique amongst siblings in the same scope
 |      :param name: Docs at Terraform Registry: {@link https://www.terraform.io/docs/providers/aws/d/lb.html#name DataAwsLb#name}.
 |      :param tags: Docs at Terraform Registry: {@link https://www.terraform.io/docs/providers/aws/d/lb.html#tags DataAwsLb#tags}.
 |      :param count: 
 |      :param depends_on: 
 |      :param lifecycle: 
 |      :param provider:
```

**Comment by jsteinich** [comment]
I tested out setting the `id` property in Terraform on a resource where it wasn't mentioned in the documentation. The result was `Error: Invalid or unknown key`. As such, I'm quite hesitant to just add `id` everywhere the output schema says we should.

I checked in a Terraform plugin for an ide (JetBrains product) and it seemed to suggest `id` only when correctly available. I'm not sure how it is getting that information, but perhaps there's a similar method that could be employed here.

Ideally, the Terraform schema command wouldn't show `id` as assignable when it isn't.
